### DISCUSSION TITLE
Change rotation of OLED SH1106

### DISCUSSION THREAD

**Discussion by Fabesson** [outdoorbits/little-backup-box#203] [Q&A]
I wanted to glue an OLED SH1106 into a transparent raspberry case. But due to dimensions of the oled an the constuction of the case it is only possible to glue in one position.The screen would now be upside down.
Is it possible to rotate the screen 180 degrees?
I already searched in the file "/var/www/little-backup-box/display.py"
But couldn't find anything regarding the rotation.

A really nice project so far. I am intensively testing for my first trip with GoPro and little-backup-box.

**Comment by outdoorbits** [comment]
Rotating is not implemented yet but I would expect, it's not a big deal. I'm a bit busy until Sunday, but I think I could upload this function until Monday or Tuesday next week.

**Reply by Fabesson** [reply]
oh you were a quick moment faster.

**Comment by Fabesson** [comment]
I found a line in display.py which contains the rotation of the screen.
 self.device.capabilities(self.conf_DISP_RESOLUTION_X,self.conf_DISP_RESOLUTION_Y,0,mode=self.conf_DISP_COLOR_MODEL)
The third option "0" in this case is the rotation of the screen.
I now changed it to "2" for 180 degrees rotation.
Testing successful so far. Only a reboot required.
Maybe it is possible to set the rotation in the web ui?
This would be easier and the rotation had not  to be set via command-line after an update.

**Comment by outdoorbits** [comment]
Thank you, good job. I'll implement just this.

**Comment by outdoorbits** [comment]
I just uploaded it. Could you have a look?

**Reply by Fabesson** [reply]
I updated my little Backup Box and tested both options (0° and 180°).
It works without any problems.
Thank you for the quick solution.
